### DISCUSSION TITLE
ustrftime

### DISCUSSION THREAD

**Discussion by iyassou** [micropython/micropython#12133] [Show and tell]
Hello 👋

I wanted to share my [implementation of `time.strftime`](https://github.com/iyassou/ustrftime). It accepts `time.localtime()` tuples and supports most formatting directives (except timezone-related ones).

**Comment by rkompass** [comment]
Did you note that there is a time.strftime in micropython lib's time module? It has not all your formatting directives however.
I started to program strftime myself until I found that.

**Reply by iyassou** [reply]
No, I was not aware of its existence. Reading it now I also wasn't aware you could pass tuples to `micropython.const`, neat.
Thanks.

**Comment by bidrohini** [comment]
Thanks for sharing.
